book/src/contributing/using-dev-container.md

```diff
@@ -0,0 +1,55 @@
+# Development in Dev Containers
+
+```admonish note
+This document is a heavy work in progress and may not contain sufficient information.
+```
+
+[Dev Containers](https://containers.dev/) are a way to package a number of
+"developer tools" (compilers, bundlers, package managers, loaders, etc.) into a
+single object. This is helpful when many people want to contribute to a project:
+each person only has to install the (single) Dev Container on their own machine
+to start working. By definition, the Dev Container has a consistent set of tools
+that are known to work together. This avoids a fuss with finding the proper
+versions of each of the build tools.
+
+To use a Dev Container on your local computer with VS Code, you must install the
+[VS Code Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+extension and its system requirements.
+
+When you first start the Dev Container, the build process may take a long time
+(as much as 20-40 minutes) as the container collects all the packages of the
+full PRQL toolchain. Once it's running, though, startup is fairly fast.
+
+## How you use it
+
+1. Clone the git repo onto your hard drive as usual. Then you start the Dev
+   Container (say, using VS Code) that bundles all the developer tools.
+2. Edit files locally (say, using VS Code) to work on PRQL. The tools in the Dev
+   Container watch for changed files and rebuild the project as needed.
+3. When you are satisfied with the changes, you can commit them to the repo, and
+   push the changes as usual.
+
+## Starting and stopping the Dev Container
+
+_Instructions needed - both from VS Code and maybe from CLI?._
+
+## Using PRQL in a Dev Container
+
+After the Dev Container starts up, the VS Code Terminal pane shows a Dev
+Container command line. Enter one of the following commands to work on a
+component of PRQL.
+
+VS Code offers a link to "Open in Browser" to the proper port to see the results
```

Comment:
FYI I wasn't familiar with this, and couldn't test it yet — as long as someone is confident then great

Reply:
I believe VS Code detects when some program uses the port in the container and issues a notification and forwards it to the host machine port. (I don't understand the details of how it works!)

In any case, this will be obvious if anyone use it, so there is little need to specify it here.

@richb-hanover Any thoughts?

Reply:
I say, leave out that "open in browser" comment, in the spirit of retaining only what's always going to be true, and not better explained elsewhere.